algorithmic_efficiency/logging_utils.py

```diff
@@ -0,0 +1,492 @@
+from datetime import datetime
+import glob
+import json
+import os
+import platform
+import re
+import subprocess
+from typing import Any, Optional
+
+from absl import flags
+from absl import logging
+import GPUtil
+import pandas as pd
+import psutil
+
+from algorithmic_efficiency import spec
+
+FLAGS = flags.FLAGS
```

Comment:
in general I try to avoid using flags within any library code, and instead just collect them all in main() (or one function down the call stack) and then pass them around as arguments. if flags are directly read throughout library code, it can lead to non-obvious behavior where the reader is not aware of how flag values are always used.

it seems like the extra logging data and eval freq are the only flags used, so they could probably be passed into the constructor or a function call as arguments.

Reply:
Done. Thanks!